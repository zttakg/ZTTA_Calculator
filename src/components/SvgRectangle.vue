<template>
  <rect
    :x="positionX"
    :y="positionY"
    :rx="radiusX"
    :ry="radiusY"
    :width="width"
    :height="height"
    :fill="fillColor"
    :stroke-width="stroke"
    :stroke="strokeColor"
  />
</template>

<script>
export default {
  name: "SvgRectangle",

  props: {
    width: {
      type: String,
      default: "300"
    },
    height: {
      type: String,
      default: "300"
    },
    rx: String,
    ry: String,
    x: {
      type: String,
      default: "0"
    },
    y: {
      type: String,
      default: "0"
    },
    radius: String,
    strokeWidth: {
      type: Number,
      default: 0
    },
    strokeColor: String,
    fillColor: String
  },

  computed: {
    radiusX() {
      const rx = this.rx && this.ry ? this.rx : this.radius;
      return rx > 0 ? rx : undefined;
    },
    radiusY() {
      const ry = this.ry && this.ry ? this.ry : this.radius;
      return ry > 0 ? ry : undefined;
    },
    positionX() {
      const x = this.x
        ? this.strokeWidth + parseInt(this.x, 10)
        : this.strokeWidth;
      return x > 0 ? x : undefined;
    },
    positionY() {
      const y = this.y
        ? this.strokeWidth + parseInt(this.y, 10)
        : this.strokeWidth;
      return y > 0 ? y : undefined;
    },
    stroke() {
      return parseInt(this.strokeWidth, 10) > 0 ? this.strokeWidth : undefined;
    }
  }
};
</script>
