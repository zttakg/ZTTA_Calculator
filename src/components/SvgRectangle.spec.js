import { shallowMount } from "@vue/test-utils";
import SvgRectangle from "./SvgRectangle";

describe("<SvgRectangle /> component", () => {
  it("should render correct without props", () => {
    const wrapper = shallowMount(SvgRectangle);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render all props", () => {
    const wrapper = shallowMount(SvgRectangle, {
      propsData: {
        width: "200",
        height: "200",
        radius: "10",
        strokeWidth: 5,
        strokeColor: "#000",
        fillColor: "#000"
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render `rx` and `ry` props manually", () => {
    const wrapper = shallowMount(SvgRectangle, {
      propsData: {
        rx: "20",
        ry: "10",
        radius: "100"
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
