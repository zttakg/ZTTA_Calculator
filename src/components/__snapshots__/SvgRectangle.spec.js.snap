// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SvgRectangle /> component should render \`rx\` and \`ry\` props manually 1`] = `<rect rx="20" ry="10" width="300" height="300"></rect>`;

exports[`<SvgRectangle /> component should render all props 1`] = `<rect x="5" y="5" rx="10" ry="10" width="200" height="200" fill="#000" stroke-width="5" stroke="#000"></rect>`;

exports[`<SvgRectangle /> component should render correct without props 1`] = `<rect width="300" height="300"></rect>`;
